server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -153,6 +155,12 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             return new TermInSetQuery(name(), bytesRefs);
         }
 
+        @Override
+        public ValuesSourceType getValuesSourceType() {
+            // TODO: should this even exist? Is aggregating on the ID field valid?
```

Comment:
Possible yes, encouraged not necessarily.  See warning on: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html

> The value of the _id field is also accessible in aggregations or for sorting, but doing so is discouraged as it requires to load a lot of data in memory. In case sorting or aggregating on the _id field is required, it is advised to duplicate the content of the _id field in another field that has doc_values enabled.